atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/compact/SweepHistoryProvider.java

```diff
@@ -19,16 +19,23 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
 import com.palantir.atlasdb.schema.generated.SweepPriorityTable;
+import com.palantir.atlasdb.schema.generated.SweepPriorityTable.SweepPriorityNamedColumn;
 import com.palantir.atlasdb.schema.generated.SweepTableFactory;
 import com.palantir.atlasdb.transaction.api.Transaction;
 
 class SweepHistoryProvider {
```

Comment:
Should we move this into the store class?

Reply:
This is used by the `CompactPriorityCalculator` (which is used in Oracle at least to determine whether to try to run shrink space etc. on tables) - the idea was that you want to compact something only after you've swept it since the last compaction, because you'll have tombstones that you know can be dropped (based on the number of cells deleted).

If the only user of the sweep history was the background sweeper I'd say it probably should be merged, but given this usage it may be cleaner this way.